### ISSUE TITLE
Machine no longer advertising as exit node upon reboot

### ISSUE THREAD

**Issue by schopenhauer** [tailscale/tailscale#2383]
Is there a way to make the `--advertise-exit-node` flag persistent, as it seems like the `tailscale` daemon starts normally, but omits this flag when a machine is rebooted, which results in an exit node no longer working.

**Comment by bradfitz** [comment]
What version? What's your Tailscale IP?

**Comment by schopenhauer** [comment]
Linux/1.10.1 running on IP 100.104.47.32

**Comment by bradfitz** [comment]
Thanks, @schopenhauer.

@apenwarr, logs suggest it's no longer saving preferences in the LocalBackend.Start when the new opts.UpdatePrefs is non-nil. That is, the new UpdatePrefs code you added never had a path that saved those prefs from a Start.

This is probably the same issue as #2321 etc.

To see what's going on:

```
$ logprint  -i be024d8dba9d920716ded47912c3753ed3a765d45c2c551e7675cd9dbf72ace3 --since=100h | grep -i -E 'Program starting|routes=|prefs|control: Host|health|Switching|EditPref' | perl -npe 's/control: HostInfo.*RoutableIPs.(.+)/YES: $1/; s/control: HostInfo.(.*)/NO: $1/' | less
```

Then look at the first transition from NO to YES (where a HostInfo.RoutableIPs appears) and then look at the few log lines (without the grep) immediately before that first YES line.

@apenwarr, you want to take this?

**Comment by bradfitz** [comment]
Yes, confirmed that this is the same as #2321, so closing this in favor of that earlier one.
